### PR TITLE
Bug 1822666: authentication operator should handle operand namespace creation

### PR DESCRIPTION
**Pull request by sallyom**
Migrated operand ns and dependent resources to auth-operator managed rather than cvo.  Also, I added the yaml check that openshift-apiserver has. 
 This PR moves from manifest to bindata:
* openshift-authentication ns yaml
* operand clusterrolebinding yaml
* operand serviceaccount yaml
* branding-secret yaml 
* operand cabundle configmap yaml

### CODE REVIEW

File: pkg/operator2/operator.go [hunk 1]
```diff
@@ -122,11 +126,12 @@ func NewAuthenticationOperator(
 
 		oauthClientClient: oauthClientClient.OAuthClients(),
 
-		services:    kubeClient.CoreV1(),
-		endpoints:   kubeClient.CoreV1(),
-		secrets:     kubeClient.CoreV1(),
-		configMaps:  kubeClient.CoreV1(),
-		deployments: kubeClient.AppsV1(),
+		kubeClientHolder: resourceapply.NewKubeClientHolder(kubeClient),
```
Comment (stlaz) [inline pkg/operator2/operator.go:129]: you'll want to wire in the informers for the new resources

File: pkg/operator2/operator.go [hunk 2]
```diff
@@ -239,6 +244,19 @@ func (c *authOperator) handleSync(ctx context.Context, operatorConfig *operatorv
 	// will cause a redeploy anyway
 	// TODO move this hash from deployment meta to operatorConfig.status.generations.[...].hash
 	resourceVersions := []string{}
+	// ensure operand namespace and resources exist
+	directResourceResults := resourceapply.ApplyDirectly(c.kubeClientHolder, c.recorder, assets.Asset,
+		"oauth-openshift/ns.yaml",
+		"oauth-openshift/cabundle.yaml",
```
Comment (stlaz) [inline pkg/operator2/operator.go:250]: nit: remove the `release.openshift.io/create-only: "true"` from the file, it's no longer needed when you do this

Comment (sallyom) [reply pkg/operator2/operator.go:250]: done

File: pkg/operator2/operator.go (hunk 1 above)
Comment (sallyom) [reply pkg/operator2/operator.go:129]: the configmap added is the `v4-0-config-system-trusted-ca-bundle` and the branding secret is `v4-0-config-system-ocp-branding-template` so those are covered by the `getPrefixFilter()` function, right? https://github.com/openshift/cluster-authentication-operator/blob/master/pkg/operator2/operator.go#L170-#L171

File: pkg/operator2/operator.go (hunk 2 above)
Comment (sallyom) [reply pkg/operator2/operator.go:250]: done

File: pkg/operator2/operator.go (hunk 1 above)
Comment (stlaz) [reply pkg/operator2/operator.go:129]: yep, those should already be covered :)

File: bindata/oauth-openshift/authentication-clusterrolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: system:openshift:openshift-authentication
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin  # TODO fix, this is madness
```
Comment (stlaz) [inline bindata/oauth-openshift/authentication-clusterrolebinding.yaml:7]: could you please change the comment to: oauth-server needs to read secrets in all namespaces so there is no point trying to tighten its permissions

File: pkg/operator2/operator.go (hunk 1 above)
Comment (stlaz) [reply pkg/operator2/operator.go:129]: but you are still missing an informer for the NS of the operand, the rolebinding and the SA

File: bindata/oauth-openshift/authentication-clusterrolebinding.yaml (hunk 3 above)
Comment (sallyom) [reply bindata/oauth-openshift/authentication-clusterrolebinding.yaml:7]: done

File: pkg/operator2/operator.go [hunk 4]
```diff
@@ -161,11 +165,14 @@ func NewAuthenticationOperator(
...
 	configNameFilter := operator.FilterByNames("cluster")
 	prefixFilter := getPrefixFilter()
+	operandNSFilter := operator.FilterByNames("openshift-authentication")
 
 	return operator.New("AuthenticationOperator2", c,
 		operator.WithInformer(routeInformer, targetNameFilter),
 		operator.WithInformer(coreInformers.Services(), targetNameFilter),
 		operator.WithInformer(kubeInformersNamespaced.Apps().V1().Deployments(), targetNameFilter),
+		operator.WithInformer(kubeInformersNamespaced.Core().V1().ServiceAccounts(), targetNameFilter),
+		operator.WithInformer(kubeInformersNamespaced.Core().V1().Namespaces(), operandNSFilter),
```
Comment (sallyom) [inline pkg/operator2/operator.go:176]: I added informer for sa and ns, but looking at our other operators (i looked at service-ca-operator, openshift-apiserver-operator, cluster-kube-controller-manager-operator), AFAICT they don't add the informer for ClusterRoleBindings()  I want to ask why before I add it here.

Comment (stlaz) [reply pkg/operator2/operator.go:176]: That's a good question, I do not know. I can see that those attempt to use kube-auto-reconciliation, which we could perhaps use here, too, I only need to see how it works.

File: pkg/operator2/operator.go (hunk 1 above)
Comment (sallyom) [reply pkg/operator2/operator.go:128]: added all but clusterrolebinding, mentioned in note below

File: bindata/oauth-openshift/serviceaccount.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: openshift-authentication
+  name: oauth-openshift
+  labels:
+    app: oauth-openshift
```
Comment (deads2k) [inline bindata/oauth-openshift/serviceaccount.yaml:7]: why the label?

File: pkg/operator2/operator.go (hunk 4 above)
Comment (sallyom) [reply pkg/operator2/operator.go]: added staticresourcecontroller instead : )

**Pull request merged**